app/services/stats/management_information_generator.rb

```diff
@@ -21,21 +21,42 @@ def call
 
     attr_reader :format
 
+    def log_error(error)
+      LogStuff.send(:error, 'MI Report generation error has occured:')
+      LogStuff.send(:error, "#{error.class} - #{error.message}")
+      LogStuff.send(:error, error.backtrace.inspect.to_s)
+    end
+
+    def log_info(message)
+      LogStuff.send(:info, message)
+    end
+
+    def headers
+      Settings.claim_csv_headers.map { |header| header.to_s.humanize }
+    end
+
+    def active_non_draft_claims
+      Claim::BaseClaim.active.non_draft
+    end
+
     # TODO: separate data retrieval from exporting the data itself
     # keeping existent behaviour for now
     def generate_new_report
+      log_info('Report generation started...')
       CSV.generate do |csv|
-        csv << Settings.claim_csv_headers.map { |header| header.to_s.humanize }
-        Claim::BaseClaim.active.non_draft.find_each do |claim|
+        csv << headers
+        active_non_draft_claims.find_each do |claim|
+          log_info("Adding claim to report with id: #{claim.id} ...")
           ClaimCsvPresenter.new(claim, 'view').present! do |claim_journeys|
             if claim_journeys.any?
-              claim_journeys.each do |claim_journey|
-                csv << claim_journey
-              end
+              log_info("Adding journey for claim: #{claim.id}")
+              claim_journeys.each { |journey| csv << journey }
             end
           end
         end
       end
+    rescue StandardError => e
+      log_error(e)
```

Comment:
You can right specs that raise an error and then "spy" on the behaviour to make sure `log_error` is called and even `LogStuff` is called with relevant arguments.

have a look at [this example](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/blob/master/spec/models/timed_state_transitions/batch_transitioner_spec.rb#L60)

Reply:
Thanks @jsugarman I'll implement these changes